poster.go

```diff
@@ -2,8 +2,36 @@ package lookout
 
 import "context"
 
+// AnalysisStatus is the status reported to the provider to
+// inform that we are performing an analysis, or that it has finished
+type AnalysisStatus int
+
+const (
+	_ AnalysisStatus = iota
+	// ErrorAnalysisStatus represents an error status
```

Comment:
could you explain what is the difference between Error and Failure in our case?

Reply:
ah. I see. Failure is when we added some comments. But do we really want to show red cross?

Reply:
The difference is not so obvious here, but in `github.Provider` you can see the descriptions. Error would be an internal error that made the processing fail; and failure is a normal processing that yielded a negative result (like a failing CI build)

```
	case lookout.ErrorAnalysisStatus:
		return "error", "There was an error during the analysis", nil
	case lookout.FailureAnalysisStatus:
		return "failure", "The analysis result was negative", nil
```

Reply:
Also, please note that the PR code only uses _pending_ and _success_. _failure_ may be used at some point in the future, but not for now.

Reply:
Thank you very much for the explanation. Yeah, it's a good idea to implement all statuses even if don't use them right now. 👍